### DISCUSSION TITLE
Curious about the move away from protobuf defintions

### DISCUSSION THREAD

**Discussion by davidraleigh** [jackskj/carta#3] [General]
I remember considering using your data mapping with proto a year ago, but by that time I'd already adopted a messy mapping of my own. What was the reasoning behind shutting down protoc-gen-map? Did you project move away from protobufs? Was protobuf limiting your definitions? Just curious. I'd love to read a medium post on why the change over to carta (or maybe just a few sentences here :)). 

Thanks!
D

**Comment by jackskj** [comment]
The core functionality which I aimed to deliver was to map complex relational data. [I also did not like the "usual suspect" solutions. ](https://github.com/jackskj/carta#comparison-to-related-projects)

protoc-gen-map included code generation as well as use of protobufs. Both are handy, but neither are necessary. This is the reason why I made the move to carta. 

Whether created by proto code generation or by hand, your go structs can be used in conjunction with carta. This was not the case with protoc-gen-map, since it assumed that you were using proto. 

I did not move away from protobufs and neither were they limiting in my data definitions(but I can think of edge cases where it could be a problem). At some point, I simply wished to map my go structs that did not need to be from a proto file.  


J
